plugin/completion.py

```diff
@@ -123,7 +123,7 @@ def is_applicable(cls, view_settings: dict) -> bool:
 
     def initialize(self) -> None:
         self.initialized = True
-        session = session_for_view(self.view, 'completionProvider')
+        session = session_for_view(self.view, 'completionProvider', self.view.sel()[0].b)
```

Comment:
Same as in color.py - is this a correct change?
`initialize` is called once per view. Is it correct to base the result here on where the cursor is?

Reply:
This is deliberate to make it obvious that we are actually doing this behind the curtains.

Reply:
But then this branch is currently confused about that as the point argument was removed in session(s)_for_view. What's the ultimate goal then?

Reply:
- session_for_view: Give me the best matching session for the current point and view
- sessions_for_view: Give me all sessions for the view

Reply:
No, I mean that the `point` argument is no longer in those but then you are adding explicit `point` argument to the callers.

So I'm not sure what your intention is. To remove argument and make point implicit or add it everywhere.

BTW. If we are adding `sel()[0].being()` everywhere then please have a shared helper for it somewhere as it's a lot of repetition and also this should be null-checking selection.

Reply:
I made the `point` arg optional.